gc/base/MemorySubSpaceFlat.cpp

```diff
@@ -459,6 +459,12 @@ MM_MemorySubSpaceFlat::collectorExpand(MM_EnvironmentBase* env, MM_Collector* re
 
 uintptr_t
 MM_MemorySubSpaceFlat::releaseFreeMemoryPages(MM_EnvironmentBase* env)
+{
+	return releaseFreeMemoryPages(env, 0);
+}
+
+uintptr_t
+MM_MemorySubSpaceFlat::releaseFreeMemoryPages(MM_EnvironmentBase* env, uintptr_t memoryType)
 {
```

Comment:
I'd make this call direction between the APIs opposite. This variant of API is more generic, and it should call the special case one under some conditions. The condition would be 'if old memory type'.
But, I'd actually convert the condition into an assert.
Overall:
This method first asserts that 'old' is set  and ignores if 'new' is set: OLD == type & OLD
Then it passes the request the other API.
The other API passes the request to its child.

Reply:
sorry, I was wrong, you were right. 
the direction of calls you used actually is more natural
but I'd like that assert here, which means the other method should pass OLD, not 0